### PR TITLE
Use separately built Python dependencies when building the Agent

### PR DESCRIPTION
**Pull request by alopezz**
### What does this PR do?

Install Python dependencies based on lockfiles available in integrations-core, from custom indices where everything is pre-built.

### Motivation

[BARX-260](https://datadoghq.atlassian.net/browse/BARX-260)

### Describe how to test/QA your changes

- Test a number of Agent integrations for each platform.

[BARX-260]: https://datadoghq.atlassian.net/browse/BARX-260?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: omnibus/config/software/datadog-agent-integrations-py3.rb [hunk 1]
```diff
@@ -421,11 +258,6 @@
...
   end
 
-  # From now on we don't need piptools anymore, uninstall its deps so we don't include them in the final artifact
-  uninstall_buildtime_deps.each do |dep|
-    command "#{python} -m pip uninstall -y #{dep}"
-  end
-
   # Patch applies to only one file: set it explicitly as a target, no need for -p
   if windows_target?
     patch :source => "remove-maxfile-maxpath-psutil.patch", :target => "#{python_3_embedded}/Lib/site-packages/psutil/__init__.py"
```
Comment (FlorentClarret) [inline omnibus/config/software/datadog-agent-integrations-py3.rb:263]: qq: will we one day move the patches to integrations-core? I saw we have a `patches` folder there but it's empty

Comment (ofek) [reply omnibus/config/software/datadog-agent-integrations-py3.rb:263]: That is the plan yes but I think in this case it's possible for us to remove the patch, I'm waiting on a response: https://github.com/giampaolo/psutil/pull/2110#issuecomment-1282890008

**Review by chouquette — COMMENTED** [review]
LGTM besides the small issue on macOS

Nice work, I can't wait to see the results of this :)

File: omnibus/config/software/datadog-agent-integrations-py2.rb [hunk 2]
```diff
@@ -36,42 +35,15 @@
...
-  excluded_packages.push(/^aerospike==/)
-  excluded_folders.push('aerospike')
-end
-
 if osx_target?
-  # exclude aerospike, new version 3.10 is not supported on MacOS yet
-  excluded_folders.push('aerospike')
-
   # Temporarily exclude Aerospike until builder supports new dependency
   excluded_packages.push(/^aerospike==/)
```
[the remark this answers is not in the document]
Comment (alopezz) [reply omnibus/config/software/datadog-agent-integrations-py2.rb:40]: Good catch, and frankly I'm surprised (worried, even) that this didn't make the MacOS build fail.

**Review by ofek — APPROVED** [review]

**Review by chouquette — APPROVED** [review]

**Review by FlorentClarret — APPROVED** [review]

**Pull request merged**